blog-posts/getting-started-gatsby-next-remix.md

```diff
@@ -0,0 +1,306 @@
+
+
+#### **Gatsby**
+
+A static generated page in Gatsby consists of two parts: A GraphQL query and the actual React component (here `BlogPage`), which must be exported as default. In the GraphQL query, the required data is fetched. In this example, the Gatsby plugin [gatsby-transformer-remark](https://www.gatsbyjs.com/plugins/gatsby-transformer-remark/) is used to get the content of the corresponding markdown file.
+In the actual component, the results of the query can then be accessed via `data`.
+
+In our Gatsby blog project, we created a file named `{markdownRemark.frontmatter__path}.tsx` to create a route for each blogpost. The important thing here is that the filename is the path of the query in curly brackets. This way the data from the markdown files determines the routes of the project.
+
+```tsx
+const BlogPost = ({data}) => {
+    return (
+        <div dangerouslySetInnerHTML={{__html: data.markdownRemark.html}}/>
+    )
+}
+
+export const query = graphql`
+    query ($id: String) {
+        markdownRemark(id: {eq: $id}) {
+            frontmatter {
+                title
+                path
+            }
+            html
+        }
+    }
+`
+
+export default BlogPost
+```
+
+
+> 💡 Alternatively, pages can also be created in `gatsby-node.js`. For this, the `createPages` API provided by Gatsby must be used. For more information on this, as well as an easy-to-understand example, see the [Gatsby documentation](https://www.gatsbyjs.com/docs/creating-and-modifying-pages/#creating-pages-in-gatsby-nodejs).
+
+
+#### **Next.js**
+
+To create a static generated page in Next.js the `async` function `getStaticPaths` has to be exported. `getStaticPaths` must return an object with the keys:
+
+- `path`: This key determines which path will be pre-rendered. In the in-code comment below the structure of `path` is given as an example.
+- `fallback`: You can set the value of fallback to `true`, `false` or `blocking`
+    - `false` → Any paths not returned by `getStaticPaths` will end up in a 404 page
+    - `true` → On the first request to a not generated path, Next.js will serve a fallback version of the page, instead of a 404 page. The HTML and JSON of this path is now generated in the background and and then the JSON is sent to the requesting browser. Now the fallback page changes into the full page. Next.js will add this path to the pre-rendered pages in order to be able to provide the already generated page for further requests.
+      This is useful if you have a page with a lot of static pages because then not all pages have to be generated at built time, which makes the built much faster.
+    - `blocking` → Same procedure as for `fallback: true`, except that there is no fallback while the HTML is being generated. There are only very special uses cases in which this makes sense, such as AMP.
+
+Remember to name the file after the relevant key from the `params` object with square brackets, in our case `[path].tsx`.
```

Comment:
Should I look at the source code? Then write this afterwards. In addition I can't find the mentioned snippet, where should the reader look at ?

Reply:
Which mentioned snippet do you mean?

Reply:
You told me the details and we concluded that a highlight in the code through prismjs would help a lot when talking about code.